homeassistant/components/tuya/alarm_control_panel.py

```diff
@@ -0,0 +1,142 @@
+"""Support for Tuya Alarm."""
+from __future__ import annotations
+
+from tuya_iot import TuyaDevice, TuyaDeviceManager
+
+from homeassistant.backports.enum import StrEnum
+from homeassistant.components.alarm_control_panel import (
+    SUPPORT_ALARM_ARM_AWAY,
+    SUPPORT_ALARM_ARM_HOME,
+    SUPPORT_ALARM_TRIGGER,
+    AlarmControlPanelEntity,
+    AlarmControlPanelEntityDescription,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import (
+    STATE_ALARM_ARMED_AWAY,
+    STATE_ALARM_ARMED_HOME,
+    STATE_ALARM_DISARMED,
+    STATE_ALARM_TRIGGERED,
+)
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers.dispatcher import async_dispatcher_connect
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from . import HomeAssistantTuyaData
+from .base import EnumTypeData, TuyaEntity
+from .const import DOMAIN, TUYA_DISCOVERY_NEW, DPCode
+
+
+class Mode(StrEnum):
+    """Alarm modes."""
+
+    ARM = "arm"
+    DISARMED = "disarmed"
+    HOME = "home"
+    SOS = "sos"
+
+
+STATE_MAPPING: dict[str, str] = {
+    Mode.DISARMED: STATE_ALARM_DISARMED,
+    Mode.ARM: STATE_ALARM_ARMED_AWAY,
+    Mode.HOME: STATE_ALARM_ARMED_HOME,
+    Mode.SOS: STATE_ALARM_TRIGGERED,
+}
+
+
+# All descriptions can be found here:
+# https://developer.tuya.com/en/docs/iot/standarddescription?id=K9i5ql6waswzq
+ALARM: dict[str, tuple[AlarmControlPanelEntityDescription, ...]] = {
+    # Alarm Host
+    # https://developer.tuya.com/en/docs/iot/categorymal?id=Kaiuz33clqxaf
+    "mal": (
+        AlarmControlPanelEntityDescription(
+            key=DPCode.MASTER_MODE,
+            name="Alarm",
+        ),
+    )
+}
+
+
+async def async_setup_entry(
+    hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
+) -> None:
+    """Set up Tuya alarm dynamically through Tuya discovery."""
+    hass_data: HomeAssistantTuyaData = hass.data[DOMAIN][entry.entry_id]
+
+    @callback
+    def async_discover_device(device_ids: list[str]) -> None:
+        """Discover and add a discovered Tuya siren."""
+        entities: list[TuyaAlarmEntity] = []
+        for device_id in device_ids:
+            device = hass_data.device_manager.device_map[device_id]
+            if descriptions := ALARM.get(device.category):
+                for description in descriptions:
+                    if (
+                        description.key in device.function
+                        or description.key in device.status
+                    ):
```

Comment:
From recent learnings, we only need to check if the status is present. If there is no status, it won't work anyways.

```suggestion
                    if description.key in device.status:
```